### PR TITLE
Ncnewst5

### PR DESCRIPTION
**Pull request by Joelymodevs**

### CODE REVIEW

File: src/Articles.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    );
+  }
+  return (
+    <div className="articles">
+      {articles.map((article) => (
+        <Link to={`/articles/${article.article_id}`} key={article.title}>
+        <div key={article.article_id} className="article-card">
+          <img
+            src={article.article_img_url}
+            alt="article-image"
```
Comment (JVal-NC) [inline src/Articles.jsx]: A different alt may be the title? Something less generic essentially

File: src/FullArticle.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    const { article_id } = useParams()
+
+    useEffect(() => {
+        getArticleById(article_id).then((data) => {
+            setArticle(data);
+            setIsLoading(false)
+        }).catch((err) => {
+            console.log(err)
+        })
+    }, [article_id])
```
Comment (JVal-NC) [inline src/FullArticle.jsx:21]: I believe we need to setLoading with true when we have anything in a dependancy array.This avoids loading being false when we are fetching the articles on the second invoation of useEffect.

We would also like mobile first design to be followed.

**Review by JVal-NC — COMMENTED** [review]
Couple comments to get you started before the full PR review tomorrow.

File: src/FullArticle.jsx [hunk 3]
```diff
@@ -16,7 +16,7 @@ const FullArticle = () => {
         }).catch((err) => {
             console.log(err)
         })
-    }, [article_id])
+    }, [article_id], setIsLoading(true))
```
Comment (Augs0) [inline src/FullArticle.jsx]: This is causing an infinite re-render at the moment and so I can't view the page. The setIsLoading should happen inside of the function. First, before the axios call to indicate were waiting, then back to false at the end of the async chain. Please fix this before continuing. This state should also be false by default.

**Review by Augs0 — CHANGES_REQUESTED** [review]

File: src/FullArticle.jsx (hunk 3 above)
Comment (Joelymodevs) [reply src/FullArticle.jsx]: apologies

**Review by Augs0 — APPROVED** [review]
Approving as we seem to finally have this ticket in the bag :D a few things to consider though:
- you have a lot of needless divs throughout your code that could be changed
- your styling files are mixed with your components. It might be worth popping these in a styles folder
- the need to click a button when on the homepage is not a great UX. You could just render your articles on the homepage or have a splash page with a menu to visit different parts of the app

**Pull request merged**